File: multiplo.java
package multiplo;

import java.util.Scanner;

/**
 *
 * @author Rafathan
 */
public class EjerJava5 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {

        int a = 0, b = 0, respuesta = 1;
        Scanner reader = new Scanner(System.in);
        boolean error = true;

        while (respuesta == 1) {
            do {
                try {
                    System.out.print("\nDame una cifra: ");
                    a = Integer.parseInt(reader.next());
                    error = false;
                } catch (Exception z) {
                    System.out.println("Opcion incorrecta.");
                    error = true;
                }
            } while (error);
            do {
                try {
                    System.out.print("Dame otra cifra y te dire si " + a + " es multiplo de ella: ");
                    b = Integer.parseInt(reader.next());
                    error = false;
                } catch (Exception x) {
                    System.out.println("Opcion incorrecta.");
                    error = true;
                }
            } while (error);

            if (a % b == 0) {
                System.out.println(a + " es multiplo de " + b);
            } else {
                System.out.println(a + " no es multiplo de " + b);
            }

            System.out.print("\n1-Realizar nueva comprobacion.\n0-Salir.\n");
            do {
                try {
                    System.out.print("Seleccion: ");
                    respuesta = Integer.parseInt(reader.next());
                    error = false;
                } catch (Exception y) {
                    System.out.println("Opcion incorrecta.");
                    error = true;
                }
                if ((respuesta  1)) {
                    System.out.println("Opcion incorrecta.");
                    error = true;
                }
            } while (error);
        }
    }
}

File: multiplo.cpp
#include <cstdlib>
#include <iostream>
using namespace std;

/*
 * 
 */
int main(int argc, char** argv) {

    int a, b;
    char respuesta = '1';
    bool error = false;

    while (respuesta == '1') {
        cout << "\nDame una cifra: ";
        cin >> a;
        cout << "Dame otra cifra y te dire si " << a << " es multiplo de ella: ";
        cin >> b;

        if (a % b == 0) cout << a << " es multiplo de " << b;
        else cout << a << " no es multiplo de " << b;

        cout << "\n\n1-Realizar nueva comprobacion.\n0-Salir.\n";
        do {
            error = false;
            cout << "Seleccion: ";
            cin >> respuesta;
            if ((respuesta != '0') && (respuesta != '1')) {
                cout << "Opcion incorrecta.\n";
                        error = true;
            }
        } while (error);
    }

    return 0;
}
